src/Renderer/ThreeExtended/FirstPersonControls.js

```diff
@@ -161,6 +165,16 @@ class FirstPersonControls extends THREE.EventDispatcher {
         }
     }
 
+    translateY(dt, sign) {
+        this.camera.position.z += sign * this.options.moveSpeed * dt / 1000;
+    }
+
+    translateYGlobe(dt, sign) {
+        // compute geodesic normale
+        const normal = this.camera.position.clone().normalize();
```

Comment:
You could declare normal, outside the function and use `copy`, instead of `clone`

Reply:
yes, good idea !